### DISCUSSION TITLE
Next.js 13 App Router + SSR Theme Switching

### DISCUSSION THREAD

**Discussion by xarunoba** [saadeghi/daisyui#2216] [Show and tell]
Hi there! I'd like to share a simple example of how I got theme switching for daisyUI to work with Next.js 13 App Router's server components.

I used this [discussion](https://github.com/vercel/next.js/discussions/47952#discussion-5046708)'s solution with modifications of my own so that it will fit with daisyUI and SSR. The theme is prerendered and uses cookies to identify the user's theme server-side.

Link:
https://github.com/Xarunoba/nextjs-daisyui-ssr

**Comment by jeremyd** [comment]
THANK YOU!!!!  I have searched long and hard for this, and I'm so happy you did this.  It's working.  All the other 'solutions' I found are terrible and don't work.
